Sources/AzureIoTMiddlewareForSwift/AzureIoT.swift

```diff
@@ -0,0 +1,318 @@
+
+    public var mqttClient: MQTTClient! = nil
+    public var mqttPublishPacketId: UInt32 = 0
+    public var mqttSubscribePacketId: UInt32 = 0
+    
+    // Section 2: Implementation of MQTTClientDelegate.
+
+    let mqttQueue = DispatchQueue(label: "a", qos: .background)
+
+    public var delegateDispatchQueue: DispatchQueue {
+        mqttQueue
+    }
+
+    public func mqttClient(_ client: MQTTClient, didReceive packet: MQTTPacket) {
+        switch packet {
+        case let packet as ConnAckPacket:
+            print("< CONNACK \(packet.returnCode)")
+            _ = azure_iot_mqtt_client_connected(&internalClient)
+        case let packet as PubAckPacket:
+            print("< PUBACK \(packet.identifier)")
+            _ = azure_iot_mqtt_client_publish_completed(&internalClient, Int32(packet.identifier))
+        case let packet as SubAckPacket:
+            print("< SUBACK \(packet.identifier) | \(packet.returnCodes)")
+            _ = azure_iot_mqtt_client_subscribe_completed(&internalClient, Int32(packet.identifier))
+        case let packet as PublishPacket:
+            print("< PUBLISH \(packet.identifier) | \(packet.topic) | \(packet.qos)")
+            print("  Payload: \(String(decoding: packet.payload, as: UTF8.self))")
+
+            var mqttMessage = mqtt_message_t()
+            mqttMessage.topic = AzSpan(text: packet.topic).toCAzSpan()
+            mqttMessage.payload = AzSpan(data: packet.payload).toCAzSpan()
+            mqttMessage.qos = mqtt_qos_t(rawValue: UInt32(packet.qos.rawValue))
+
+            _ = azure_iot_mqtt_client_message_received(&internalClient, &mqttMessage)
+        default:
+            print("Packet \(packet)")
+        }
+    }
+
+    public func mqttClient(_: MQTTClient, didChange state: ConnectionState) {
+        if state == .connected {
+        }
+        else if state == .disconnected {
+            _ = azure_iot_mqtt_client_disconnected(&internalClient)
+        }
+        print("[MQTT] \(state)")
+    }
+
+    public func mqttClient(_: MQTTClient, didCatchError error: Error) {
+        print("[MQTT] Error: \(error)")
+    }
+
+    // Section 3: Callbacks for Azure IoT Universal Middleware.
+
+    let mqttClientInitFunction: mqtt_client_init_function_t = { userContext, mqttClientConfig, mqttClientHandle in
+        print("Connecting MQTT client")
+
+        var mySelf: AzureIoT = Unmanaged<AzureIoT>.fromOpaque(userContext!).takeUnretainedValue()
+
+        let address: String = (AzSpan(span: mqttClientConfig!.pointee.address)).toString()
+        let port = Int(mqttClientConfig!.pointee.port)
+        let clientId: String = (AzSpan(span: mqttClientConfig!.pointee.client_id)).toString()
+        let username: String = (AzSpan(span: mqttClientConfig!.pointee.username)).toString()
+        let password: String = (AzSpan(span: mqttClientConfig!.pointee.password)).toString()
+
+        let caCert = "/home/ewertons/code/work/s1/azure-sdk-for-c-swift/certs/baltimore.pem"
```

Comment:
this can be removed with the `.fullVerification`. It uses the cert store.

Reply:
Removed.